gx_suite_definitions/proteomics_tmt.ipynb

```diff
@@ -0,0 +1,303 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import synapseclient\n",
+    "\n",
+    "import pandas as pd\n",
+    "import great_expectations as gx\n",
+    "\n",
+    "from agoradatatools.gx import GreatExpectationsRunner\n",
+    "\n",
+    "context = gx.get_context(project_root_dir='../src/agoradatatools/great_expectations')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Expectation Suite for Proteomics TMT Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get Example Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "syn = synapseclient.Synapse()\n",
+    "syn.login()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "proteomics_tmt_file = syn.get(\"syn32210527\").path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Validator Object on Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_json(proteomics_tmt_file)\n",
+    "nested_columns = []\n",
+    "df = GreatExpectationsRunner.convert_nested_columns_to_json(df, nested_columns)\n",
+    "validator = context.sources.pandas_default.read_dataframe(df)\n",
+    "validator.expectation_suite_name = \"proteomics_tmt\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Expectations to Validator Object For Each Column"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# uniqid\n",
+    "validator.expect_column_values_to_be_of_type(\"uniqid\", \"str\")\n",
+    "validator.expect_column_value_lengths_to_be_between(\"uniqid\", 1, 25)\n",
+    "validator.expect_column_values_to_match_regex(\"uniqid\", \"^[a-zA-Z0-9.]+?|[a-zA-Z0-9-]+$\")\n",
```

Comment:
Can you help me parse this regex? So far I have: 
Starts with any alphanumeric character or ".", must have at least 1 ("+?" signifies match as few as possible before coming to the "|" I think?), followed by "|", followed by at least one alphanumeric character or "-", end. Is that correct?

Reply:
Or wait, shouldn't the "." and "|" be escaped with "\"?

Reply:
I was basically trying to get it to check that it is a valid hgnc_symbol concatenated with a valid uniprotid separated by a |. So what I was going for is, "Any combination of alphanumeric characters and periods followed by a pipe followed by any combination of alphanumeric characters and hyphens".

I originally tried to escape the pipe, but I was getting this warning:
```
[WARNING] <>:4: DeprecationWarning: invalid escape sequence \|
```
and after removing the escape character it was still working.

Reply:
Huh, interesting! Ok then, we'll call it good!